dlt/common/storages/fsspec_filesystem.py

```diff
@@ -282,13 +283,22 @@ def glob_files(
         # this is a file so create a proper file url
         bucket_url = pathlib.Path(bucket_url).absolute().as_uri()
         bucket_url_parsed = urlparse(bucket_url)
+
     bucket_url_no_schema = bucket_url_parsed._replace(scheme="", query="").geturl()
     bucket_url_no_schema = (
         bucket_url_no_schema[2:] if bucket_url_no_schema.startswith("//") else bucket_url_no_schema
     )
     filter_url = posixpath.join(bucket_url_no_schema, file_glob)
+    is_unc_path = fs_client.protocol == "file" and "$" in filter_url
```

Comment:
I started to experiment with this place but it still does not work. all paths (bucket_url) and filter_url are normalized and $ is never there - at least for your test cases